### ISSUE TITLE
Tray Icon is displayed wrong in KDE

### ISSUE THREAD

**Issue by mhoeher** [Enrico204/Whatsapp-Desktop#27]
First of all, thanks for developing and maintaining this project :) It is really useful!

I am using this app on two systems with Fedora 26 installed, using the KDE Plasma desktop, i.e.:

* KDE Plasma version: 5.10.4
* KDE Frameworks Version 5.36.0
* Qt Version: 5.7.1

With this configuration, the tray icon shown looks weird:

![spectacle f17180](https://user-images.githubusercontent.com/1911135/29809382-3c513e9a-8c9c-11e7-9256-12923ff50ddc.png)

It seems as if only a fraction of the actual icon is shown.

**Comment by Enrico204** [comment]
Sorry for the delay, I was quite busy in this month. I'll check this ASAP :-)

**Comment by mhoeher** [comment]
No hurry, the app works just fine otherwise ;-)

Thanks for looking into this!

**Comment by Enrico204** [comment]
Hi @mhoeher , I can't reproduce this behavior: I installed Fedora KDE (default configuration) and latest Whatsapp-Desktop, and the issue is not present. There are some customization that I need to have on KDE? Thanks :-)

**Comment by ZeroBit** [comment]
In my latest KDE 5 in Archlinux it looks like
![screenshot_20171007_143200](https://user-images.githubusercontent.com/6258813/31307391-633adca2-ab6c-11e7-8df9-174daf5229cb.png)
Sure I will welcome ,ore standard look like here
![whatsapp-viber](https://user-images.githubusercontent.com/6258813/31307419-ef58a570-ab6c-11e7-8900-701a3a46fda8.png)
![whatsapp-512](https://user-images.githubusercontent.com/6258813/31307420-ef5bc7dc-ab6c-11e7-91e4-dc7e1f9d2c88.png)

**Comment by Enrico204** [comment]
I tried with both Fedora KDE and ArchLinux KDE (actually, Manjaro KDE) but I can't reproduce this issue (default config in a VM). Maybe it's fixed somehow in `master` branch. What version are you using (eg. released version 0.3.13 or `master` development version)?

**Comment by ZeroBit** [comment]
I have 0.3.13 version of your app. But the same look was also in 0.3.12
For both version I used these links
[0.3.13](https://github.com/Enrico204/Whatsapp-Desktop/releases/download/v0.3.13/WhatsApp-linux-x64.tar.gz) and[ 0.3.12 ](https://github.com/Enrico204/Whatsapp-Desktop/releases/download/v0.3.12/WhatsApp-linux-x64.tar.gz)

I use theme **Breeze** for KDE

**Comment by Enrico204** [comment]
@ZeroBit can you clone latest commits from `master` and test it?

**Comment by ZeroBit** [comment]
Unfortunately I am not so big professional to clone to build and to install from **master**

**Comment by Enrico204** [comment]
@ZeroBit ok I'll build a tar.gz for you, so you can use the same PKGBUILD way to test it as if it was on release page ;-)

**Comment by ZeroBit** [comment]
Thank you, This is very good idea

**Comment by Enrico204** [comment]
@ZeroBit try with this: https://drive.google.com/open?id=0B8u78kiRBShdR1IyYUxETHhUUU0

**Comment by Enrico204** [comment]
@ZeroBit I saw your post in #34 . I still can't reproduce this issue :-(
I'll try to switch some settings in KDE, maybe there is something (non-default) that trigger this.

**Comment by psmis2** [comment]
I also suffer from this issue with KDE and I also went ahead and cloned **master** and the issue still persists.

 OS: Fedora 
 Kernel: x86_64 Linux 4.13.4-200.fc26.x86_64
 Shell: bash 4.4.12
 Resolution: 3440x1440
 DE: KDE 5.36.0 / Plasma 5.10.5
 WM: KWin
 WM Theme: Breeze
 GTK Theme: Adwaita [GTK2], Default [GTK3]
 Icon Theme: breeze
 Font: Sans Serif 
 CPU: Intel Core i7-5960X @ 16x 3.5GHz [35.0°C]
 GPU: Gallium 0.4 on AMD FIJI (DRM 3.18.0 / 4.13.4-200.fc26.x86_64, LLVM 4.0.0)
 RAM: 4459MiB / 64337MiB

**Comment by mhoeher** [comment]
Hi @Enrico204,

sorry for the late reply, was busy that weekend.

Regarding customizations: I am not aware of anything I installed which might cause this. For me, the issue also happens independently on two different setups (my private laptop as well as my one for work). Both run the latest Fedora release with up-to-date packages.

As you mentioned you tested in a VM: One difference between running in a VM and on real hardware might be graphics support and graphics drivers. For testing, I set up a VM running Fedora and inside the VM the tray icon shows up correctly. As a side note, issue #6 does also not occur in the freshly set up VM (on both my "real" machines, a single click opens the menu instead) - however, that might be coincidence.

**Comment by Enrico204** [comment]
Thanks! So it might depend on the fact that I'm using VMs to test. I'll try to setup a real machine for testing asap.
Thanks again for testing :-)

**Comment by KottV** [comment]
Can confirm on openSUSE Tumbleweed, Plasma 5.11, Frameworks 5.38, Nvidia blob.

**Comment by sultan1988** [comment]
Same issue kde 5.11 Frameworks 5.39 QT 5.9.1

**Comment by Termuellinator** [comment]
For me, this doesn't occur - manjaro KDE, 5.10.5, installed on a physical machine via AUR - so i doubt that the VM is the difference.

**Comment by Enrico204** [comment]
I tried with Fedora, openSUSE and Manjaro (all with KDE). I can't reproduce this behavior, so this is not related to GPU (and/or the use of VMs). Maybe it's related to DPI/resolution. I'll do further testing :-)

**Comment by mhoeher** [comment]
I did some research on the topic. I guess the issue is in fact due to some inconsistencies/bugs/whatsoever between Electron and KDE, see e.g. https://github.com/vector-im/riot-web/issues/3133.

I uninstalled `libappindicator` and `libappindicator-gtk3` and restarted the app - now the tray icon is shown properly (and also left-clicking the icon causes the window to be shown in case it's hidden, instead of just showing the context menu).

**Comment by Enrico204** [comment]
Thanks!
So I think that I'll drop a line in the README file with your workaround. Then I hope that Electron guys can find a (definitive) solution on this.

**Comment by frojasaracena** [comment]
Hi. I have the same problem on Fedora 25 and 27. I can't remove `libappindicator` because is needed by slack app. Other apps based on Electron don't have this problem in KDE.

**Comment by chrisvdb** [comment]
Same issue on Kubuntu 17.10 with 5.11.4. Other Electron apps like Slack do not have this issue.

**Comment by xfactor2000** [comment]
Hi,
same issue on Manjaro Linux with KDE 5.11.5. Can't uninstall `libappindicator` and `libappindicator-gtk3` because I don't have them.

**Comment by usernatilds** [comment]
Same issue with PIA VPN https://www.reddit.com/r/kde/comments/65ywan/any_kde_users_that_use_the_private_internet/#bottom-comments

When resize PIA Icon to 24x24, work for me, now i try in whatsapp-desktop and i solve the problem with icon, see:
![screenshot_12_02_2018_as08_52_42](https://user-images.githubusercontent.com/25496130/36093548-234b8dce-0fd2-11e8-9b3c-eee5dcfc7670.png)

The command I used to solve was:
$ sudo su
# apt install imagemagick
# cd /opt/whatsapp-desktop/resources/app/assets/icon/
# mogrify -resize 24x24 *.png

Now start whatsapp-desktop.

**Comment by frojasaracena** [comment]
The solution of @usernatilds worked for me! 
Fedora 27 + KDE.

**Comment by usernatilds** [comment]
@frojasaracena, grateful for feedback.

**Comment by usernatilds** [comment]
@Enrico204 can see here? What did you think?

**Comment by Enrico204** [comment]
@usernatilds I see. While the issue still (I can't just resize icons for everyone, because that will break the icon on other D.E.), I think that I can arrange a workaround based on the environment (in other words, set the resized icon when necessary). I'll do some testing :-)
Thanks everyone for feedbacks :-)

**Comment by usernatilds** [comment]
@Enrico204, I reported the similar problem in the PIA program, they already fixed in the new version, however, I updated whatsapp-desktop and now the problem has returned, needing to do everything manually again with each new version.

**Comment by Enrico204** [comment]
Yeah, to be honest, I haven't worked on this for 0.5. I'll schedule this for 0.5.1 or 0.6, I need some time to test an Electron upgrade (just in case...)
